orderer/common/filerepo/filerepo.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hyperledger/fabric/common/configtx"
+)
+
+type Repo struct {
+	Options Options
+}
+
+type Options struct {
+	FileSuffix          string
```

Comment:
the file suffix is the extension or suffix for the files we are saving in the file repo. For joinblocks this will likely be something like `.joinblock` so all joinblocks are saved as `<filerepo dir>/<channel>.joinblock`